src/utils/fmt.zig

```diff
@@ -0,0 +1,38 @@
+const std = @import("std");
+const sig = @import("../lib.zig");
+
+inline fn maxArg(comptime field: std.builtin.Type.StructField) field.type {
+    return switch (@typeInfo(field.type)) {
+        .ComptimeInt => sig.utils.types.defaultValue(field) orelse @compileError("comptime_int field without default value has no max value"),
```

Comment:
Yeah, you deduced it, `@TypeOf(.{ 123 }) == struct { comptime u32 = 123 }`. Comptime fields must always have a default value, and must always be their default value - they're essentially glorified decls, kind of like the `.len` on an array.

Since `comptime_int` doesn't have an actual upper or lower bound, this is the only ergonomic way to get a precise buffer size for it in this case.